### PR TITLE
Disjoint_Set_Union.java

### PR DESCRIPTION
**Pull request by knockcat**
DSU TEMPLATE.

### Description
A Disjoint-Set Union (DSU) data structure, also known as a Union-Find data structure, is a data structure that helps manage a collection of disjoint sets. It provides two primary operations: union and find.

Here's a brief overview of the DSU data structure:

1. **MakeSet(x):** This operation creates a new set containing a single element 'x.' Initially, each element is in its own set.

2. **Find(x):** This operation finds the representative (also known as the leader or root) of the set to which 'x' belongs. It is used to determine if two elements belong to the same set by checking if their representatives are the same. It performs path compression, which means that as you repeatedly call Find(x) on an element, it will update the parent pointers along the path to the root, effectively flattening the tree and improving future Find operations.

3. **Union(x, y):** This operation merges the sets containing 'x' and 'y' into a single set. It does this by finding the representatives of the sets containing 'x' and 'y' using the Find operation and then making one of them the parent of the other.

The DSU data structure is commonly used for various applications, including:

- Kruskal's algorithm for finding minimum spanning trees.
- Detecting cycles in a graph.
- Connected components in a graph.
- Dynamic connectivity problems.
- Image segmentation.
- Disjoint set-based data structures like Disjoint Set Data Structures (DSDS) and Disjoint Set Forests (DSF).

DSU is efficient, especially when using path compression and union-by-rank heuristics. The time complexity for the Find operation is nearly constant on average, making it a valuable tool for solving problems that involve partitioning elements into disjoint sets.

### Related/Fixes Issue
[Reference the related issue or task using its number or link.]

### Changes Made
[Briefly list the specific code changes made.]

### Screenshots (if applicable)
[Include any relevant screenshots or images here.]

- [x] Fixed Bug/issues
- [x] Added algo/questions_answers/docs
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Review by rohan472000 — COMMENTED** [review]
provide brief about this DSU and give wikipedia link also

**Review by rohan472000 — CHANGES_REQUESTED** [review]
change file name to DisjointSetUnion.java

**Review by knockcat — COMMENTED** [review]
sir i have made the changes,
sir the description is added in pull request description.

**Comment by rohan472000** [comment]
image
I can't see any proper description.

File: Graphs/DisjointSetUnion.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import java.io.*;
+import java.util.*;
+class DisjointSet {
```
Comment (rohan472000) [inline Graphs/DisjointSetUnion.java:3]: add brief about DSU and wiki link under comments here.

**Review by rohan472000 — CHANGES_REQUESTED** [review]

**Comment by knockcat** [comment]
sir is this fine now.

File: Graphs/DisjointSetUnion.java (hunk 1 above)
Comment (knockcat) [reply Graphs/DisjointSetUnion.java:9]: added

**Comment by rohan472000** [comment]
Thanks for your contribution.

**Comment by knockcat** [comment]
Thankyou sir for telling changes and reviewing it so calmly.

**Pull request merged**